addons/binding/org.openhab.binding.fritzaha/src/main/java/org/openhab/binding/fritzaha/internal/hardware/FritzahaWebInterface.java

```diff
@@ -0,0 +1,304 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.fritzaha.internal.hardware;
+
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.openhab.binding.fritzaha.config.FritzAhaBridgeConfiguration;
+import org.openhab.binding.fritzaha.internal.hardware.callbacks.FritzAhaCallback;
+import org.openhab.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class handles requests to a Fritz!OS web interface for interfacing with
+ * AVM home automation devices. It manages authentication and wraps commands.
+ * 
+ * @author Robert Bausdorf, Christian Brauers 
+ * @since 2.0
+ */
+public class FritzahaWebInterface {
+
+	/**
+	 * Configuration of the bridge from {@link org.openhab.bindig.fritzaha.handler.FritzAhaBridgeHandler}
+	 */
+	protected FritzAhaBridgeConfiguration config;
+	/**
+	 * Current session ID
+	 */
+	protected String sid;
+	/**
+	 * HTTP client for asynchronous calls
+	 */
+	protected HttpClient asyncclient;
+	/**
+	 * Maximum number of simultaneous asynchronous connections
+	 */
+	protected int asyncmaxconns = 20;
+	/**
+	 * Event publisher used by binding
+	 */
+	// protected EventPublisher eventPublisher;
+
+	static final Logger LOG = LoggerFactory
+			.getLogger(FritzahaWebInterface.class);
+	// Uses RegEx to handle bad FritzBox XML
+	/**
+	 * RegEx Pattern to grab the session ID from a login XML response
+	 */
+	protected static final Pattern SID_PATTERN = Pattern
+			.compile("<SID>([a-fA-F0-9]*)</SID>");
+	/**
+	 * RegEx Pattern to grab the challenge from a login XML response
+	 */
+	protected static final Pattern CHALLENGE_PATTERN = Pattern
+			.compile("<Challenge>(\\w*)</Challenge>");
+	/**
+	 * RegEx Pattern to grab the access privilege for home automation functions
+	 * from a login XML response
+	 */
+	protected static final Pattern ACCESS_PATTERN = Pattern
+			.compile("<Name>HomeAuto</Name>\\s*?<Access>([0-9])</Access>");
+
+	/**
+	 * This method authenticates with the Fritz!OS Web Interface and updates the
+	 * session ID accordingly
+	 * 
+	 * @return New session ID
+	 */
+	public String authenticate() {
```

Comment:
Throughout this method there should be no warn or error logging, but you will need to find a way to set the thing status on the BridgeHandler (see https://github.com/eclipse/smarthome/blob/master/docs/sources/architecture/thing-status.md#status-detail-mapping).